### DISCUSSION TITLE
Underwhelming codespaces single threaded performance

### DISCUSSION THREAD

**Discussion by enricopolanski** [community/community#48959] [Codespaces]
### Select Topic Area

Product Feedback

### Body

Hello, just wanted to give a small feedback.

I have recently started working with a repository with very heavy single threaded requirements.

The project is a huge typescript monorepo, on top of that we use a custom [typescript compiler](https://github.com/ts-plus/) and we do heavy code transformations and generation with eslint.

The project is absolutely heavy, there's no doubt on that, most pcs are going to struggle at least a bit, and the overall editor experience isn't great with intellisense feedback and compilation lasting a lot of time. The cleanest experience is offered running at least an m2 mac book pro which gives almost instant experience.

That being said, I have been particularly underwhelmed by single threaded performance of codespaces. It seems to be more or less on par with a desktop cpu like a 3200g I currently have (the experience is in the same tier of performance). Scaling to more cores from 4 to 16 seems to do quite little.

Thus my feedback. Is Github planning to provide codespaces (even for a bonus tier) that have much better single threaded performance?

Right now we're limited in scaling our team as fundamentally we can't use codespaces for development and we have to invest several thousand euros for each worker to get a better machine.

Bests, Enrico.

**Comment by thomas-sickert** [comment]
Thanks for the feedback, we'll keep track of it but we don't have any immediate plans to add new machine types
